From 203eb317533a59835f98bc18e69b2c49d0af2bd5 Mon Sep 17 00:00:00 2001
From: Fancy Fang <chen.fang@nxp.com>
Date: Wed, 16 Aug 2017 16:38:20 +0800
Subject: [PATCH 2383/5242] MLK-16197-8 video: fbdev: dcss: move subsam config
 to probe

commit  70bf351273f573b5d62debaeb237e9fd2d7c2c1f from
https://source.codeaurora.org/external/imx/linux-imx.git

Move the SUBSAM config to probe stage to generate
display timings as soon as the graphic layer unblank.

Signed-off-by: Fancy Fang <chen.fang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/imx_dcss.c |   10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/drivers/video/fbdev/mxc/imx_dcss.c b/drivers/video/fbdev/mxc/imx_dcss.c
index 0a3ca33..72f4544 100644
--- a/drivers/video/fbdev/mxc/imx_dcss.c
+++ b/drivers/video/fbdev/mxc/imx_dcss.c
@@ -2784,12 +2784,6 @@ static int dcss_blank(int blank, struct fb_info *fbi)
 	if (blank == FB_BLANK_UNBLANK) {
 		/* dcss output timings can only be set for fb0 */
 		if (!fb_node) {
-			ret = dcss_subsam_config(info);
-			if (ret) {
-				dev_err(&pdev->dev, "subsam config failed\n");
-				goto out;
-			}
-
 			ret = dcss_dtg_config(fb_node, info);
 			if (ret) {
 				dev_err(&pdev->dev, "dtg config failed\n");
@@ -3255,6 +3249,10 @@ static int dcss_probe(struct platform_device *pdev)
 	/* enable encoder if exists */
 	dcss_enable_encoder(info);
 
+	ret = dcss_subsam_config(info);
+	if (ret)
+		goto unregister_fb0;
+
 	/* register channel 1: video */
 	ret = dcss_register_one_ch(1, info);
 	if (ret)
-- 
1.7.9.5

